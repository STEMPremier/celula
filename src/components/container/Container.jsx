import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './container.less';

/**
 * `<Containers />` group componets together, adding margin, or padding, (or neither) and an optional border.
 */
const Container = ({ border, children, gap }) => {
  const classes = cx('ce-container', {
    'ce-container__border': border,
    [`ce-container__${gap}`]: gap !== 'none',
  });

  return <div className={classes}>{children}</div>;
};

Container.propTypes = {
  /**
   * Add a border around the `<Container />`.
   */
  border: PropTypes.bool,
  /**
   * Any renderable elements the `<Container />` is to contain.
   */
  children: PropTypes.node.isRequired,
  /**
   * The method the spacing (if any) around the `<Container />` is applied.
   */
  gap: PropTypes.oneOf(['margin', 'padding', 'none']),
};

Container.defaultProps = {
  border: false,
  gap: 'margin',
};

export default Container;
