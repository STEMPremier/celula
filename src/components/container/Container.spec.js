/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Container from '.';

describe('<Container />', () => {
  describe('render()', () => {
    test('render the Container component', () => {
      const wrapper = shallow(
        <Container>
          <span>This is stuff in a container.</span>
        </Container>,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
/* eslint-enable react/jsx-filename-extension */
