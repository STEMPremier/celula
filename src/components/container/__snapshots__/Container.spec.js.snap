// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> render() render the Container component 1`] = `
<div
  className="ce-container ce-container__margin"
>
  <span>
    This is stuff in a container.
  </span>
</div>
`;
